### PR TITLE
Add Setup script for OpenDebugAD7

### PR DESCRIPTION
**Pull request by WardenGnaw**
This PR adds a Setup script and List files for scenarios that launch OpenDebugAD7.exe.
The scenarios are VS Code C++ extension or VS CodeSpaces CMake Debug.

This script simplifies the process of patching the files.

### CODE REVIEW

File: tools/Setup.csx [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+        System.Environment.Exit(1);
+    }
+
+    public void ParseArguments(IList<string> args)
+    {
+        if (args.Count == 0)
+        {
+            PrintHelp();
+        }
```
Comment (gregg-miskelly) [inline tools/Setup.csx:55]: >} [](start = 8, length = 1)

return

File: tools/Setup.csx [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        foreach (string arg in args)
+        {
+            if (arg.StartsWith("-") || arg.StartsWith("/"))
+            {
+                switch(arg.Substring(1).ToLower())
+                {
+                    case "help":
+                    case "?":
+                        PrintHelp();
+                        break;
```
Comment (gregg-miskelly) [inline tools/Setup.csx:66]: >break [](start = 24, length = 5)

return;

File: tools/Setup.csx [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+            File.Copy(srcPath, destPath, overwrite: true);
+        }
+    }
+
+} 
+
+Setup setup = new Setup();
+setup.ParseArguments(Args);
+setup.Run();
```
Comment (gregg-miskelly) [inline tools/Setup.csx:228]: >setup [](start = 0, length = 5)

Don't continue if this printed the help.

File: tools/Setup.csx [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+                        break;
+                    case "debug":
+                        Configuration = Configuration.Debug;
+                        break;
+                    case "release":
+                        Configuration = Configuration.Release;
+                        break;
+                    default:
+                        Console.WriteLine(string.Format("Unknown flag {0}", arg));
+                        break;
```
Comment (gregg-miskelly) [inline tools/Setup.csx:81]: >break [](start = 24, length = 5)

throw, or otherwise don't continue processing.

File: tools/Setup.csx [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+    public void Run()
+    {
+        string scriptDirectoryPath = Path.GetDirectoryName(GetSourceFile());
+        string srcDirectoryPath = Path.GetFullPath(Path.Join(scriptDirectoryPath, "..", "src"));
+        string binDirectoryPath = Path.GetFullPath(Path.Join(scriptDirectoryPath, "..", "bin"));
+
+        // No client flag provided, try to auto detect.
+        if (Client == Client.None)
+        {
```
Comment (gregg-miskelly) [inline tools/Setup.csx:164]: Should this just be required to specify?

File: tools/Setup.csx [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+
+    public Setup()
+    {
+        Client = Client.None;
+        Configuration = Configuration.Debug;
+    }
+
+    private void PrintHelp()
+    {
+        Console.WriteLine("USAGE: dotnet script Setup.csx -- <Target-Path> [-vs|-vscode] [-debug|-release]");
```
Comment (gregg-miskelly) [inline tools/Setup.csx:40]: ><Target-Path> [](start = 61, length = 13)

For vscode, do we want to have a default path?

**Review by gregg-miskelly — APPROVED** [review]
Otherwise LGTM

File: tools/Setup.csx (hunk 5 above)
Comment (WardenGnaw) [reply tools/Setup.csx:164]: I think auto resolving on target path is acceptable. Going to add automatically finding the cpptools extension path if they want VS Code.

**Pull request merged**